### ISSUE TITLE
SQL File Reload

### ISSUE THREAD

**Issue by JonathonRichardson** [WNPRC-EHR-Services/lkpm#3]
SQL queries scripts should be reloaded by default.  For some reason, in LKPM modules, that is not occurring, and LabKey needs to be restarted to pick up changes to query scripts.

This is super annoying to develop against.

**Comment by JonathonRichardson** [comment]
Alright, so, after looking into this, I found that the problem is that Docker mounts the volumes using CIFS when using Docker for Windows, which [doesn't propagate remote changes to files](http://stackoverflow.com/questions/8476419/java-watchservice-not-generating-events-while-watching-mapped-drives) to the local **inotify** service (see [here](http://stackoverflow.com/questions/8124617/getting-file-create-notifications-for-cifs-mount-in-linux) for another reference on this) .  Oracle is aware of the issue, but I doubt [that ticket](http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6965508) will go anywhere, as it hasn't been updated for over two years at the time I'm writing this.  Docker [is also aware](https://github.com/docker/for-win/issues/56) of this limitation.

All of this was for Docker for Windows, but I also remember seeing this behavior on my work mac (which uses fuse.osxfs rather than CIFS), but I'll have to wait until to Monday to check it again there to confirm that the root cause is also that `WatchService` is not triggering `QueryBasedModuleResourceCache.StandardListener#entryModified()`, which updates the cached values.

It seems that the best solution (although far from perfect) would be to recursively touch the files in question every 10 seconds or so from within the container using a Quartz job, which will properly trigger the `WatchService`.  This should only be done for modules that are not in production mode and that have their source files available.  There are a couple ways I could do this:
 - Include the code in core LabKey server code.  In order to affect other deployments, it would probably be best to only schedule the poller when in dev mode and some variable is set (e.g. `-DpollModuleFiles=true`).
 - Have `DBUtils` register the poller on startup.
    - This would exceed the scope of that module in my opinion.
 - Create a new module that basically just does this.
    - Should other customizations become necessary for docker, this could house those changes, such as augmenting the classpath to dynamically recompile JSPs in dev environments (right now this code is hackily in WebUtils).

I need to mull this over for a little.  For now, the workaround is just to use `docker-compose exec labkey /bin/bash` to enter the container and just touch the files manually to trigger them to reload.

**Comment by JonathonRichardson** [comment]
The workaround command is as follows:

```
cd /usr/local/labkey/modules
find /LabKey/sources/build/explodedModule/ /LabKey/sources/lkpm/modules/*/build/explodedModule/ -exec touch {} \;
```

Also, make sure you 
- [ ] Don't touch the module file itself, which would cause LabKey to try and reload the module itself
- [ ] Update lkpm-watch.  Right now, the touch triggers it to kick off some tasks.

**Comment by JonathonRichardson** [comment]
I ended up creating a `FileToucher` background task in **DBUtils** that spins up in dev mode.  It is set to run every 3 seconds, and doesn't seem to consume too much CPU.  It does take a couple seconds to find the files, yielding a lag of up to 10 seconds from file edit to the environment recognizing the change.  Using `find` on the command line would be faster, but I opted for a simple to code, pure Java solution.  There is room for improvement in the algorithm, but it works for now.  The commit in question is 56206e32e0b201c1db3391cb13e35ade0eda7477.
